src/app/components/shared/detail-view/render-field/render-field.component.ts

```diff
@@ -89,8 +90,11 @@ export class RenderFieldComponent extends WithUnsubscribe()
     super();
   }
 
-  public ngOnInit(): void {
-    this.ngOnChanges();
+  public ngOnInit() {
+    // This is required because unit tests do not run ngOnChanges
+    if (!this.model && !this.display && !this.children) {
+      this.ngOnChanges();
+    }
```

Comment:
link to angular bug?

Otherwise it seems like there should be a shim in the *test helpers* that patches this... not the other way around

Reply:
Its not a bug, but a "feature" of unit tests where assigning values to a component does not trigger the ngOnChanges event detecting even when calling `detectChanges()`. You have to create a host component which wraps the test component and passes the variables through. Ive implemented that now so that this is no longer needed